### PR TITLE
MPDataRetrieval: add support for "bandstructure", etc

### PR DESCRIPTION
**Pull request by albalu**
## Summary

add support for "bandstructure", "phonon_bandstructure", "phonon_ddb", "phonon_bandstructure", "phonon_dos" with the help of get_*_by_material_id functions to address #258

### CODE REVIEW

File: matminer/data_retrieval/retrieve_MP.py [hunk 1]
```diff
@@ -24,19 +25,39 @@ def __init__(self, api_key=None):
...
-            same as in get_data
+            criteria (dict): the same as in get_data
+            properties ([str]): the same properties supported as in get_data
+                plus: "structure", "initial_structure", "final_structure",
+                "bandstructure", "phonon_bandstructure", "phonon_ddb",
+                "phonon_bandstructure", "phonon_dos". Note that for a long list
+                of compounds, it make take a long time to retain some of these
+                objects.
+            index_mpid (bool): the same as in get_data
+            line_mode (bool): the bandstructure returned in line mode (True) or
```
Comment (computron) [inline matminer/data_retrieval/retrieve_MP.py]: I would remove this ``line_mode`` argument, it is too specific to a single data type.

Maybe just have the properties be "bandstructure" for uniform and "bandstructure_line" or "bandstructure:line" or something similar for line-mode.

**Comment by computron** [comment]
Thanks! I added just one minor comment to address.

**Comment by computron** [comment]
Thanks!

**Comment by albalu** [comment]
@computron , I added the bandstructure_uniform option consistent with MPRester terminology (bandstructure==line mode band structure); I made this clear in docs.

**Pull request merged**